test/GoodGhosting.test.js

```diff
@@ -106,13 +106,22 @@ contract("GoodGhosting", (accounts) => {
             await approveDaiToContract(player);
             await goodGhosting.makeDeposit({ from: player });
         }
+        const totalWinners = await goodGhosting.winners.length;
+        if (totalWinners > 0) {
+            const latestWinner = await goodGhosting.winners(new BN(totalWinners).sub(new BN(1)));
+            assert(latestWinner === player);
+        }
+
         // accounted for 1st deposit window
         // the loop will run till segmentCount - 1
         // after that funds for the last segment are deposited to protocol then we wait for segment length to deposit to the protocol
         // and another segment where the last segment deposit can generate yield
         await timeMachine.advanceTime(weekInSecs);
+
         await goodGhosting.depositIntoExternalPool({ from: player1 });
         await timeMachine.advanceTime(weekInSecs);
+        await truffleAssert.reverts(goodGhosting.makeDeposit({ from: player }), " Deposit available only between segment 1 and segment n-1 (penultimate)");
+
```

Comment:
imo, we should not add `asserts` inside a "helper" function. it does not list the test case that we're testing. I suggest we create specific test cases for these cases, removing them from here.